icepyx/core/icesat2data.py

```diff
@@ -727,21 +727,21 @@ def download_granules(self, path, verbose=False, subset=True, restart=False, **k
         Beginning download of zipped output...
         Data request [##########] of x order(s) is complete.
         """
-     
+
         # if not os.path.exists(path):
         #     os.mkdir(path)
         # os.chdir(path)
 
         if not hasattr(self, '_granules'): self.granules
-        
+
         if restart == True:
             pass
         else:
             if not hasattr(self._granules, 'orderIDs') or len(self._granules.orderIDs)==0: self.order_granules(verbose=verbose, subset=subset, **kwargs)
-    
-        self._granules.download(verbose, path, session=self._session, restart=restart)
-  
-   
+
+        self._granules.download(verbose, path, session=self._session, restart=restart, **kwargs)
```

Comment:
This implementation for `convert` will cause issues given the way the code is set up right now (which may point to a weakness in the existing code rather than this being a bad way to implement this functionality - it will need some more thought). Specifically, acceptable kwargs here (as noted in the docstring) are ONLY those that are accepted by the NSIDC subsetter. The APIformatting module relies on the proper set of passed kwargs to format the request properly.